ip.cpp:148: error: dereferencing pointer ‘ss.52’ does break strict-aliasing rules

I can't build libzmq on Ubuntu Lucid but I don't have this problem on Ubuntu Precise.

It's not critical for me. I'll use it on Precise.

``` sh
bbigras@hostname:~/libzmq⟫ make
Making all in src
make[1]: Entering directory `/home/bbigras/libzmq/src'
make  all-am
make[2]: Entering directory `/home/bbigras/libzmq/src'
  CXX    libzmq_la-address.lo
  CXX    libzmq_la-clock.lo
  CXX    libzmq_la-ctx.lo
  CXX    libzmq_la-curve_client.lo
  CXX    libzmq_la-curve_server.lo
  CXX    libzmq_la-devpoll.lo
  CXX    libzmq_la-dist.lo
  CXX    libzmq_la-epoll.lo
  CXX    libzmq_la-err.lo
  CXX    libzmq_la-fq.lo
  CXX    libzmq_la-io_object.lo
  CXX    libzmq_la-io_thread.lo
  CXX    libzmq_la-ip.lo
cc1plus: warnings being treated as errors
ip.cpp: In function ‘int zmq::get_peer_ip_address(zmq::fd_t, std::string&)’:
ip.cpp:148: error: dereferencing pointer ‘ss.52’ does break strict-aliasing rules
ip.cpp:148: note: initialized from here
make[2]: *** [libzmq_la-ip.lo] Error 1
make[2]: Leaving directory `/home/bbigras/libzmq/src'
make[1]: *** [all] Error 2
make[1]: Leaving directory `/home/bbigras/libzmq/src'
make: *** [all-recursive] Error 1
```

libzmq e376ad2787ebef91748921f1fb4766ce2b49d6a7
Ubuntu 10.04.4 LTS

--- a/src/ip.cpp
+++ b/src/ip.cpp
@@ -145,7 +145,14 @@ int zmq::get_peer_ip_address (fd_t sockfd_, std::string &ip_addr_)
         return 0;
 
     ip_addr_ = host;
-    return (int) ((struct sockaddr *) &ss)->sa_family;
+
+    union {
+        struct sockaddr sa;
+        struct sockaddr_storage sa_stor;
+    } u;
+
+    u.sa_stor = ss;
+    return (int) u.sa.sa_family;
 }
 
 void zmq::set_ip_type_of_service (fd_t s_, int iptos)

